signup/signupController.js

```diff
@@ -106,19 +110,41 @@ define('signupController', [], function() {
 				"siteId": $scope.sharedProperties.selectedSite
 			};
 
-			SignupService.storeAppointments(data).then((response) => {
-				if (typeof response !== 'undefined' && response && response.success){
-					document.body.scrollTop = document.documentElement.scrollTop = 0;
-					$scope.appointmentId = response.appointmentId;
-					$scope.successMessage = $sce.trustAsHtml(response.message);
+			const firstName = $scope.data.firstName || '';
+			const lastName = $scope.data.lastName || '';
+			SignupService.getExistingClientAppointments(firstName, lastName).then((response) => {
+				if (typeof response !== 'undefined' && response && response.success) {
+					// response.numberExistingAppointments === false when query is empty because none exist
+					if(response.numberExistingAppointments === false || response.numberExistingAppointments['numberExistingAppointments'] === '0') {
+						$scope.hasExistingAppointment = false;
+					} else {
+						$scope.hasExistingAppointment = true;
+					}
```

Comment:
The above suggestion will simplify this code even more, but it can be simplified as written too:
```suggestion
					// response.numberExistingAppointments === false when query is empty because none exist
					$scope.hasExistingAppointment = response.numberExistingAppointments !== false && response.numberExistingAppointments['numberExistingAppointments'] !== '0';
```